### PR TITLE
Show values in tabular comparison according to indicator choropleth type & colour code using profile config

### PR DESCRIPTION
**Pull request by goyal1092**
## Description
Tabular comparison according to choropleth method.
Changes: 

* choropleth method subindicator should display %
* On hover of choropleth method subindicator show tooltip with absolute value
* Color coding according to chart config in profile

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-643

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally
- [x] 👩‍🎨 does the design match the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out
- [x] commit messages are meaningful

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers

### Testing

- [x]  ✅ added (appropriate) unit tests
- [x]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

File: __tests__/gui/tabular_comparison/tabular_comparison.js [hunk 1]
```diff
@@ -66,7 +66,12 @@ Then('I check result table after adding second indicator', (type) => {
    cy.get(`[data-testid=result-table]`).find('th').eq(2).should('have.text', 'Employment status : Unemployed');
    cy.get(`[data-testid=result-table]`).find('tbody tr').eq(0).find('td').eq(0).should('have.text', 'Eastern Cape');
    cy.get(`[data-testid=result-table]`).find('tbody tr').eq(0).find('td').eq(1).should('have.text', '301,909');
-   cy.get(`[data-testid=result-table]`).find('tbody tr').eq(0).find('td').eq(2).should('have.text', '952,709');
+   cy.get(`[data-testid=result-table]`).find('tbody tr').eq(0).find('td').eq(1).should('have.css', 'background-color').and('eq', 'rgb(124, 162, 206)')
+   cy.get(`[data-testid=result-table]`).find('tbody tr').eq(0).find('td').eq(2).should('have.text', '17.02%');
+   cy.get(`[data-testid=result-table]`).find('tbody tr').eq(0).find('td').eq(2).should('have.css', 'background-color').and('eq', 'rgb(124, 162, 206)')
+   cy.get(`[data-testid=result-table]`).find('tbody tr').eq(0).find('td').eq(2).trigger('mouseover');
+   cy.get(`[role="tooltip"]`).first().find('div').should('have.text', '952,709');
+   cy.get(`[data-testid=result-table]`).find('tbody tr').eq(0).find('td').eq(2).trigger('mouseout');
```
Comment (michaelglenister) [inline __tests__/gui/tabular_comparison/tabular_comparison.js:74]: Is it useful to trigger the mouseout even here? Should we then check that the tooltip is hidden?

**Review by michaelglenister — COMMENTED** [review]
One small question in the test, but it's not a blocker

The changes look good and have test coverage thanks

File: __tests__/gui/tabular_comparison/tabular_comparison.js (hunk 1 above)
Comment (goyal1092) [reply __tests__/gui/tabular_comparison/tabular_comparison.js:74]: Initially i just did mouse over for both of cols but second tooltip didn't work correctly. so i used mouse out event to simulate proper behaviour.
We are using tooltip of MUI so i don't think we need to test showing and hiding of tooltip.

Comment (goyal1092) [reply __tests__/gui/tabular_comparison/tabular_comparison.js:74]: @michaelglenister

**Review by michaelglenister — APPROVED** [review]

File: src/js/defaultValues.js [hunk 2]
```diff
@@ -8,7 +10,18 @@ let chartConfiguration = {
     defaultType: 'Percentage', // [Value|Percentage]
     xTicks: null
 };
+
+const choroplethConfig = {
```
Comment (emre2038) [inline src/js/defaultValues.js:14]: I didn't understand this, why did we add this here @goyal1092 ?

Comment (goyal1092) [reply src/js/defaultValues.js:14]: Required default color ranges if there was no config in profile @emre2038 
Also used if profile is working with older config

File: src/js/tabular_comparison/result.js [hunk 3]
```diff
@@ -48,9 +50,20 @@ const Result = (props) => {
...
+          return [maxScaleValue * -1, maxScaleValue]
+        }
+        return [d3min(values), d3max(values)]
+    }
+
     const setBackgroundColors = (rows) => {
-        let darkestColor = '#BABABA';
-        let lightestColor = '#ffffff';
+        let choroplethConfig = props.profileConfig?.choropleth || {};
+        choroplethConfig = fillMissingKeys(choroplethConfig, defaultValues.choroplethConfig || {});
```
Comment (goyal1092) [inline src/js/tabular_comparison/result.js:66]: `defaultValues.choroplethConfig` Getting used here @emre2038

File: src/js/defaultValues.js (hunk 2 above)
Comment (emre2038) [reply src/js/defaultValues.js:14]: why can't we use the defaults from 
https://github.com/OpenUpSA/wazimap-ng-ui/blob/948b34cbd996bcbda1c0764679a47cfc91f928b4/src/js/configurations/geography_sa.js#L220

Comment (goyal1092) [reply src/js/defaultValues.js:14]: I didn't realise there was more then one config file.
In tabular comparison we were using defaultValues.chartConfiguration.
Is there a specific reason we didn't move chartConfiguration in geography_sa.js ?

Comment (emre2038) [reply src/js/defaultValues.js:14]: I think the initial purpose was that, defaultValue.js would store the constant values which do not change at all. but geography_sa.js stores the default values for the configurable variables. we can override those values by profile configs. so we need to use the one in geography_sa.js

**Review by michaelglenister — APPROVED** [review]
Changes to the default config and title look good and have tests. Thanks
